### PR TITLE
#5772 nsfw changes in gallery

### LINKED ISSUE #5772: NSFW tag/content changes
since we do have some [nsfw](https://deploy-preview-5768--koda-canary.netlify.app/rmrk/collection/C46E1CCC37647EC061-NEO) in the app, It would be nice to handle it somehow. Mostly because the images appear on the landing page quite often.
Related issues: 
- #5762
- #5455
- #285

all the changed components are in handoff file, page NSFW tag: https://www.figma.com/file/JXNtryaNYhY0XJNArzvPQx/Handoff?node-id=125%3A997&t=957PEWGiLnUmryLs-1

A few changes will be made to improve this:

### Cards
- cards will get blurred dark opacity overlay 
- also icon indicating nsfw content - this icon is for example used on bigger social networks like instagram and twitter, so I think it won't be an issue 
image
- hover works the same, the change is only the overlay over them - the content on hover still has overlay like other nft cards, that means if you hover over the light mode one, the img is slightly brighter, its basically stacking two overlays over top of each other (see prototype in figma)
- the blurred/opacity overlay is same for light and dark mode 
- blurr set to 80
image
- opacity 15% 
image

### Gallery item detail
- this will be same for dark mode and light mode
image
- see the blur settings in figma
- higher opacity to make the text more legible (25%) 
- the "show content" serves as button, once this button is clicked, overlay disappears
- full screen button is disabled when you have the blur overlay over 
- we can maybe change the copy, do you have any ideas? 

### Settings
- By default, nft marked as nsfw will appear normally everywhere, with a blurred overlay.
- The user should be able to disable the display of all nsfw.
- can you please add option to settings: "hide all content tagged with nsfw" 
- It will be basic on/off button, we can use the old one till we redesign it
image
- @JustLuuuu also mentioned that the option should not change back when you disconnect. It does now, issue #5555 
- please add id to the general tab. 

### Report
- reporting not tagged nsfw content - the option is already in gallery item detail
- I will make design for different types of report and link the issue - nsfw could be first

### Other
- I would not personally show filter in explore to "hide nsfw content" - user will have this option in settings
- open to suggestion, lmk if we should change something else

### PR DESCRIPTION
**Pull request by prachi00**
**Thank you for your contribution** to the [KodaDot - One Stop Shop for Polkadot NFTs](https://kodadot.xyz).

👇 __ Let's make a quick check before the contribution.

## PR Type

- [ ] Bugfix
- [x] Feature
- [ ] Refactoring

## Context

- [x] Partially Related #5772
- [ ] Requires deployment <snek/rubick/worker>

#### Before submitting pull request, please make sure:

- [x] My contribution builds **clean without any errors or warnings**
- [x] I've merged recent default branch -- **main** and I've no conflicts
- [x] I've tried to respect high code quality standards
- [x] I've didn't break any original functionality

#### Optional

- [ ] I've tested it at </ksm/collection>
- [ ] I've tested PR on mobile
- [ ] I've written unit tests 🧪
- [ ] I've found edge cases

#### Had issue bounty label?

- [x] Fill up your KSM address: 
[Payout](https://beta.kodadot.xyz/transfer/?target=EzGc4s9PgCPx1YnF3fqzhLzVHpHMTL4LWPScwpDrR8JKgSU)

#### Community participation

- [x] [Are you at KodaDot Discord?](https://discord.gg/35hzy2dXXh)

## Screenshot 📸

- [x] My fix has changed **something** on UI; a screenshot is best to understand changes for others.
Screen Shot 2023-05-01 at 8 42 24 PM

### <samp>🤖 Generated by Copilot at 4aab639</samp>

This pull request adds support for NSFW media items in the gallery, by blurring them and allowing the user to toggle the blur. It modifies the `GalleryItem`, `MediaItem`, and `GalleryItemDescription` components to use the `NSFW` trait as a prop and an icon.

### <samp>🤖 Generated by Copilot at 11e5d62</samp>

> _Blinded by the lewdness, you can't see the truth_
> _`isLewd` is the judge, the traitor and the proof_
> _Reveal the hidden content, with a click of fate_
> _But beware of the `console.log`, it might be too late_

### CODE REVIEW

**Comment by prachi00** [comment]
Leftovers:

- [ ] Adding "hide all content tagged with nsfw" in settings, I am not sure how would we filter this in explore
- [ ] NSFW Cards - again not sure where to get nsfw values for cards since it isnt done yet for nftentities

**Comment by yangwao** [comment]
@kodadot/qa-guild

File: components/gallery/GalleryItem.vue [hunk 1]
```diff
@@ -30,6 +31,7 @@
             :title="nftMetadata?.name"
             is-detail
             :original="isMobile"
+            :is-n-s-f-w="galleryDescriptionRef?.isNSFW"
```
Comment (vikiival) [inline components/gallery/GalleryItem.vue:34]:
```suggestion
            :is-lewd="galleryDescriptionRef?.isNSFW"
```

**Review by vikiival — CHANGES_REQUESTED** [review]
please use lewd instead of nsfw in code

**Comment by degenofdotsama** [comment]
so this doesn't blur the home page preview but only if you click on an NFT?   

it works fine when clicking on a NSFW NFT but does nothing 
Screenshot 2023-05-02 at 07 57 48

Screenshot 2023-05-02 at 07 57 18

**Comment by prachi00** [comment]
> so this doesn't blur the home page preview but only if you click on an NFT?
> 
> it works fine when clicking on a NSFW NFT but does nothing Screenshot 2023-05-02 at 07 57 48
> 
> Screenshot 2023-05-02 at 07 57 18

yeah for explore we need https://github.com/kodadot/rubick/issues/173 and https://github.com/kodadot/loligo/issues/22 to clear first

File: libs/ui/src/components/MediaItem/MediaItem.vue [hunk 2]
```diff
@@ -94,6 +115,35 @@ export default {
...
       }
     },
+    showContent() {
+      this.isLewdBlurredLayer = false
+    },
   },
 }
 </script>
+
+<style lang="scss" scopednv>
```
Comment (floyd-li) [inline libs/ui/src/components/MediaItem/MediaItem.vue:125]: hmmm...scopednv?

File: pages/transfer.vue [hunk 3]
```diff
@@ -260,6 +260,10 @@ export default class Transfer extends mixins(
 
   protected onAmountFieldChange() {
     /* calculating usd value on the basis of price entered */
+    console.log(
+      'this.fiatStore.getCurrentKSMValue',
+      this.fiatStore.getCurrentKSMValue
+    )
```
Comment (floyd-li) [inline pages/transfer.vue:266]: remove console.log pls

**Review by floyd-li — CHANGES_REQUESTED** [review]

File: libs/ui/src/components/MediaItem/MediaItem.vue (hunk 2 above)
Comment (prachi00) [reply libs/ui/src/components/MediaItem/MediaItem.vue]: oops

File: components/collection/activity/ownerInsightsTabs/FlipperTab.vue [hunk 4]
```diff
@@ -98,11 +99,14 @@ import { format } from '@/components/collection/activity/utils'
 import NFTsDetaislDropdown from './NFTsDetaislDropdown.vue'
 import { timeAgo } from '@/components/collection/utils/timeAgo'
 import { Flippers } from '@/composables/collectionActivity/types'
+import { calculateUsdFromKsm } from '~~/utils/calculation'
+import { useFiatStore } from '~~/stores/fiat'
```
Comment (vikiival) [inline components/collection/activity/ownerInsightsTabs/FlipperTab.vue:103]:
```suggestion
import { calculateUsdFromKsm } from '@/utils/calculation'
import { useFiatStore } from '@/stores/fiat'
```

**Review by vikiival — CHANGES_REQUESTED** [review]

File: components/collection/activity/ownerInsightsTabs/FlipperTab.vue [hunk 5]
```diff
@@ -130,6 +134,11 @@ const isFlipperMoreNFTSectionOpen = flippers.value.reduce(
   }),
   {}
 )
+
+const usdValue = computed(() =>
+  calculateUsdFromKsm(100000, fiatStore.getCurrentKSMValue as number)
+)
```
[the remark this answers is not in the document]
Comment (prachi00) [reply components/collection/activity/ownerInsightsTabs/FlipperTab.vue]: oh, seems like my changes from another issue got merged here, just checked the whole pr and removed those, sorry about that

File: components/gallery/GalleryItemDescription.vue [hunk 6]
```diff
@@ -141,6 +141,13 @@ const parent = computed(() => {
     return useGalleryItem(nft.value?.parent?.id)
   }
 })
+const isLewd = computed(() => {
+  return properties.value?.find((item) => {
+    return item.trait_type === 'NSFW'
+  })
+})
```
Comment (Jarsen136) [inline components/gallery/GalleryItemDescription.vue:148]:
```suggestion
const isLewd = computed(() => {
  return Boolean(properties.value?.find((item) => {
    return item.trait_type === 'NSFW'
  }))
})
```

**Review by Jarsen136 — APPROVED** [review]
code lgtm

**Review by floyd-li — APPROVED** [review]

**Comment by prachi00** [comment]
can we merge this?

**Review by vikiival — APPROVED** [review]

**Comment by vikiival** [comment]
pay 45 usd

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $45 @ 26.01 USD/KSM ~ 1.73 $KSM
🧗 [EzGc4s9PgCPx1YnF3fqzhLzVHpHMTL4LWPScwpDrR8JKgSU](https://kusama.subscan.io/account/EzGc4s9PgCPx1YnF3fqzhLzVHpHMTL4LWPScwpDrR8JKgSU)
🔗 [0x93d2e570a32ec205a3f73ed52ea55bc68c3af4c89ccd6f3f1234b9d770105d39](https://kusama.subscan.io/extrinsic/0x93d2e570a32ec205a3f73ed52ea55bc68c3af4c89ccd6f3f1234b9d770105d39)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Pull request merged**